src/packages/core/components/icon/icon.element.ts

```diff
@@ -16,29 +16,48 @@ export class UmbIconElement extends UmbLitElement {
 	@state()
 	private _color?: string;
 
+	@state()
+	private _fallbackColor?: string;
+
+	/**
+	 * Color alias or a color code directly.\
+	 * If a color has been set via the name property, this property will override it.
+	 * */
 	@property({ type: String })
 	public set color(value: string) {
-		if (!value) return;
-		this.#setColorStyle(value);
+		if (value) {
+			this.#setColorStyle(value, true);
+		} else {
+			this._color = undefined;
+		}
 	}
 	public get color(): string {
 		return this._color ?? '';
 	}
 
-	#setColorStyle(value: string) {
+	#setColorStyle(value: string, dominant = false) {
 		const alias = value.replace('color-', '');
 		const variable = extractUmbColorVariable(alias);
-		this._color = alias ? (variable ? `--uui-icon-color: var(${variable})` : `--uui-icon-color: ${alias}`) : undefined;
+		const color = alias ? (variable ? `--uui-icon-color: var(${variable})` : `--uui-icon-color: ${alias}`) : undefined;
+		if (dominant) {
+			this._color = color;
+		} else {
+			this._fallbackColor = color;
+		}
 	}
 
+	/**
+	 * The icon name. Can be appended with a color.\
+	 * Example **icon-heart color-red**
+	 */
 	@property({ type: String })
 	public set name(value: string | undefined) {
-		const [icon, alias] = value ? value.split(' ') : [];
+		const [icon, color] = value ? value.split(' ') : [];
 
-		if (alias) {
-			this.#setColorStyle(alias);
+		if (color) {
+			this.#setColorStyle(color);
 		} else {
-			this._color = undefined;
+			this._fallbackColor = undefined;
```

Comment:
I think you don't need to explicitly set this to undefined if there is no color. The undefined value is the default of `_fallbackColor` anyway.

Reply:
We also need to be able to reset the color value, if the color is removed again